server/lib/src/be/mod.rs

```diff
@@ -571,6 +571,10 @@ pub trait BackendTransaction {
                 filter_error!("Requested a top level or isolated AndNot, returning empty");
                 (IdList::Indexed(IDLBitRange::new()), FilterPlan::Invalid)
             }
+            FilterResolved::Invalid(_) => {
+                // Partial is the only one that doesn't run the risk of being cancelled due to resource constraint
+                (IdList::Partial(IDLBitRange::new()), FilterPlan::Invalid)
```

Comment:
I think this should be IdList::Indexed, since Invalid can never match, so we don't need to trigger a filter test when this term returns. The issue with Partial is if this returns Partial and it's in an Or clause, then the neighbour in the Or condition also becomes partial and forces filter testing. So this assertion being "never matches" can be Indexed so that the index state of the other IdList's can be used.

Reply:
Indexed can cause the query to be cancelled due to overuse of resources however, Which is why I opted to go with the Partial. I'll trust your judement on this one. just thought it would be better to ensure this doesn't push the query ove the limit